src/cosmos/bundle.ts

```diff
@@ -0,0 +1,384 @@
+      export const v1beta1 = { ..._53, ..._54, ..._216, ..._230 };
+    }
+    export const v1beta1 = { ..._55, ..._56 };
+  }
+  export namespace capability {
+    export const v1beta1 = { ..._57, ..._58, ..._59, ..._60 };
+  }
+  export namespace crisis {
+    export const v1beta1 = { ..._61, ..._62, ..._63, ..._64, ..._243, ..._253 };
+  }
+  export namespace crypto {
+    export const ed25519 = { ..._65, ..._66 };
+    export const multisig = { ..._67, ..._68 };
+    export const secp256k1 = { ..._69, ..._70 };
+    export const secp256r1 = { ..._71, ..._72 };
+  }
+  export namespace distribution {
+    export const v1beta1 = {
+      ..._73,
+      ..._74,
+      ..._75,
+      ..._76,
+      ..._77,
+      ..._78,
+      ..._79,
+      ..._80,
+      ..._217,
+      ..._231,
+      ..._244,
+      ..._254,
+    };
+  }
+  export namespace evidence {
+    export const v1beta1 = {
+      ..._81,
+      ..._82,
+      ..._83,
+      ..._84,
+      ..._85,
+      ..._86,
+      ..._87,
+      ..._88,
+      ..._218,
+      ..._232,
+      ..._245,
+      ..._255,
+    };
+  }
+  export namespace feegrant {
+    export const v1beta1 = {
+      ..._89,
+      ..._90,
+      ..._91,
+      ..._92,
+      ..._93,
+      ..._94,
+      ..._95,
+      ..._96,
+      ..._219,
+      ..._233,
+      ..._246,
+      ..._256,
+    };
+  }
+  export namespace genutil {
+    export const v1beta1 = { ..._97, ..._98 };
+  }
+  export namespace gov {
+    export const v1beta1 = {
+      ..._99,
+      ..._100,
+      ..._101,
+      ..._102,
+      ..._103,
+      ..._104,
+      ..._105,
+      ..._106,
+      ..._220,
+      ..._234,
+      ..._247,
+      ..._257,
+    };
+  }
+  export namespace mint {
+    export const v1beta1 = { ..._107, ..._108, ..._109, ..._110, ..._111, ..._112, ..._221, ..._235 };
+  }
+  export namespace params {
+    export const v1beta1 = { ..._113, ..._114, ..._115, ..._116, ..._222, ..._236 };
+  }
+  export namespace slashing {
+    export const v1beta1 = {
+      ..._117,
+      ..._118,
+      ..._119,
+      ..._120,
+      ..._121,
+      ..._122,
+      ..._123,
+      ..._124,
+      ..._223,
+      ..._237,
+      ..._248,
+      ..._258,
+    };
+  }
+  export namespace staking {
+    export const v1beta1 = {
+      ..._125,
+      ..._126,
+      ..._127,
+      ..._128,
+      ..._129,
+      ..._130,
+      ..._131,
+      ..._132,
+      ..._133,
+      ..._134,
+      ..._224,
+      ..._238,
+      ..._249,
+      ..._259,
+    };
+  }
+  export namespace tx {
+    export namespace signing {
+      export const v1beta1 = { ..._135, ..._136 };
+    }
+    export const v1beta1 = { ..._137, ..._138, ..._139, ..._140, ..._225, ..._239 };
+  }
+  export namespace upgrade {
+    export const v1beta1 = { ..._141, ..._142, ..._143, ..._144, ..._226, ..._240 };
+  }
+  export namespace vesting {
+    export const v1beta1 = { ..._145, ..._146, ..._147, ..._148, ..._250, ..._260 };
+  }
+}
```

Comment:
Nice. This was requested here: https://github.com/stephenh/ts-proto/issues/212 and https://github.com/cosmos/cosmjs/issues/647.

What's the reason for using namespaces (I don't understand them TBH). Can't you just create a nested object?

I'm not sure if it is convenient to intoduce index files here silently. This cuases a situation where the same symbol can be imports in various ways. I'd prefer to keep that step separate.

Reply:
- [ ] https://github.com/osmosis-labs/telescope/issues/177

Reply:
So namespaces are basically equivalent to nested objects when it transpiles. We could create nested objects, this just made it a bit easier to reason about the code. We can add options to the `bundle` option to make changes here and also eliminate the index files completely for the first PR.